File: src/main/java/com/lalkalol/mujahid/audio/LoopMode.java
package com.lalkalol.mujahid.audio;

public enum LoopMode {
    OFF,
    TRACK,
    QUEUE;

    public static LoopMode from(String value) {
        for (LoopMode mode : values()) {
            if (mode.name().equalsIgnoreCase(value)) {
                return mode;
            }
        }
        return null;
    }
}

File: src/main/kotlin/com/lalkalol/mujahid/audio/LoopMode.kt
package com.lalkalol.mujahid.audio

enum class LoopMode {
    OFF,
    TRACK,
    QUEUE,
    ;

    companion object {
        fun from(value: String): LoopMode? =
            entries.firstOrNull { it.name.equals(value, ignoreCase = true) }
    }
}
